### PR TITLE
Change status filter to buttons

### PR DESCRIPTION
**Pull request by kntsoriano**
Closes https://github.com/exercism/v3-project-management/issues/47.

### CODE REVIEW

**Comment by iHiD** [comment]
Looks good to me but I'll leave it to @ErikSchierboom to check/merge on Tuesday 🙂

File: app/javascript/components/student/tracks-list/StatusFilter.jsx [hunk 1]
```diff
@@ -1,22 +1,26 @@
 import React from 'react'
 
-export function StatusFilter({ options, dispatch }) {
-  function handleChange(e) {
-    dispatch({ type: 'status.changed', payload: { status: e.target.value } })
+export function StatusFilter({ options, dispatch, ...props }) {
```
Comment (ErikSchierboom) [inline app/javascript/components/student/tracks-list/StatusFilter.jsx:3]: To me the `props` argument implies that we don't necessarily care about the props' individual values but just about all the props. What if we would do:

```suggestion
export function StatusFilter({ options, dispatch, value }) {
  const currentValue = value || options[0].value
```

This still passes the tests (well, if we use `{option.value === currentValue}` later on).

File: app/javascript/components/student/tracks-list/StatusFilter.jsx [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
-        })}
-      </select>
+      {options.map((option) => {
+        return (
+          <button
+            key={option.value}
+            disabled={option.value === value}
+            onClick={(e) => {
+              handleChange(e, option.value)
+            }}
```
Comment (ErikSchierboom) [inline app/javascript/components/student/tracks-list/StatusFilter.jsx:18]:
```suggestion
            onClick={(e) => handleChange(e, option.value)}
```

**Review by ErikSchierboom — APPROVED** [review]

**Comment by kntsoriano** [comment]
@iHiD Ready to merge.

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
@kntsoriano There is a merge conflict.

**Pull request merged**